providers/provider_default.go

```diff
@@ -162,14 +162,13 @@ func (p *ProviderData) CreateSessionStateFromBearerToken(ctx context.Context, ra
 
 	newSession := &sessions.SessionState{
 		Email:             claims.Email,
-		User:              claims.Email,
+		User:              claims.Subject,
```

Comment:
Let me know if you think we should move this to a different PR (since this required changing a test in `oauth2proxy_test.go` potentially changing expected behavior by users).

This now aligns this to how the OIDC `CreateSessionStateFromBearerToken` provider would've extracted claims to sessions.

Reply:
Agreed, this is a bug, it's `claims.Subject` in the OIDC provider so happy to leave it like this for now. Since 499 hasn't made it into a release yet, no need to consider this breaking

Reply:
So if I recall correctly, 499 mostly moved this code verbatim from oauth2proxy.go to the default_provider.

The previous versions also had `User: claims.Email` (hence my question about this fix needing a note that this is breaking)